src/frontend-samples/display-styles-sample/DisplayStylesUI.tsx

```diff
@@ -0,0 +1,106 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { IModelApp, IModelConnection, ScreenViewport, Viewport } from "@bentley/imodeljs-frontend";
+import "common/samples-common.scss";
+import { ControlPane } from "Components/ControlPane/ControlPane";
+import { ReloadableViewport } from "Components/Viewport/ReloadableViewport";
+import * as React from "react";
+import DisplayStylesApp from "./DisplayStylesApp";
+import { DisplayStyle } from "./Styles";
+import { Toggle } from "@bentley/ui-core";
+
+interface DisplayStylesUIState {
+  activePresetIndex: number;
+  merge: boolean;
+  viewport?: Viewport;
+}
+
+interface DisplayStylesUIProps {
+  iModelName: string;
+  iModelSelector: React.ReactNode;
+  displayStyles: DisplayStyle[];
+}
+
+const CUSTOM_STYLE_INDEX = 0;
+
+export default class DisplayStylesUI extends React.Component<DisplayStylesUIProps, DisplayStylesUIState> {
+
+  public state: DisplayStylesUIState = { activePresetIndex: 2, merge: false };
+
+  // Called by the control and will update the active display style.
+  private readonly _onChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const index = Number.parseInt(event.target.value, 10);
+    this.setState({ activePresetIndex: index });
+  }
+
+  // Called by the control and updates wether to also apply the Custom display style.
+  private readonly _onToggle = (isOn: boolean) => {
+    this.setState({ merge: isOn });
+  }
+
+  // Will be triggered once when the iModel is loaded.
+  private readonly _onIModelReady = (_iModel: IModelConnection) => {
+    const vp = IModelApp.viewManager.selectedView;
+    if (undefined === vp)
```

Comment:
Is this test really needed?  Other samples just always use view passed to onViewOpen.  Ex. Heatmap

Reply:
Probably not.  Without it, it does (theoretically) create a race condition.  Will remove for consistency.